MIDAS/src/systems.cpp

```diff
@@ -165,6 +168,21 @@ DECLARE_THREAD(kalman, RocketSystems* arg) {
     }
 }
 
+/**
+ * This thread will handle the firing of pyro channels when the FSM sets the pyro flags
+*/
+DECLARE_THREAD(pyro, RocketSystems* arg) {
+
+    while (true) {        
+        FSMState current_state = arg->rocket_data.fsm_state.getRecent();
+        Pyro new_pyro = arg->sensors.pyro.tick_upper(current_state, arg->rocket_data.orientation.getRecent());
+        arg->rocket_data.pyro.update(new_pyro);
+        THREAD_SLEEP(16);
```

Comment:
why 16 ms sleep time

Reply:
I was told 60hz refresh time for this thread, it should be a trivial change later down the line